lib/puppet/file_system/windows.rb

```diff
@@ -177,14 +200,17 @@ def set_dacl(path, dacl)
     Puppet::Util::Windows::Security.set_security_descriptor(path, new_sd)
   end
 
-  def secure_dacl(current_sid)
+  def secure_dacl(current_sid, owner_permission = FULL_CONTROL, group_permission = FULL_CONTROL)
     dacl = Puppet::Util::Windows::AccessControlList.new
     [
      Puppet::Util::Windows::SID::LocalSystem,
      Puppet::Util::Windows::SID::BuiltinAdministrators,
      current_sid
     ].uniq.map do |sid|
       dacl.allow(sid, FULL_CONTROL)
+      permission = owner_permission
+      permission = group_permission if Puppet::Util::Windows::SID::BuiltinAdministrators
```

Comment:
Is this supposed to be conditional? I think right now `BuiltinAdministrators` is always defined, so `permission` is overwritten with `group_permission`.